runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/stream/TcpClientFactory.java

```diff
@@ -311,10 +314,12 @@ private void onNetConnected()
             }
         }
 
-        private void onNetRejected()
+        private void onNetRejected(
+            InetSocketAddress remoteAddress)
         {
             final long traceId = supplyTraceId.getAsLong();
-
+            String address = remoteAddress == null ? null : remoteAddress.toString();
+            event.remoteAccessFailed(traceId, routedId, address);
```

Comment:
Let's avoid converting to `String` and back to `StringFW` if feasible.

Reply:
The source is InetSocketAddress.toString() here, no StringFW's are involved at this point.